src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1923,12 +1923,12 @@ void TorrentImpl::handleFileRenameFailedAlert(const lt::file_rename_failed_alert
 
 void TorrentImpl::handleFileCompletedAlert(const lt::file_completed_alert *p)
 {
-    const int fileIndex = m_torrentInfo.nativeIndexes().indexOf(p->index);
-    Q_ASSERT(fileIndex >= 0);
-
     qDebug("A file completed download in torrent \"%s\"", qUtf8Printable(name()));
```

Comment:
@glassez is this necessary? commenting this out gives me 10% on top of this PR.

Reply:
> @glassez is this necessary?
commenting this out gives me 10% on top of this PR.

Do "release" builds affected?

Reply:
Yes, now I'm testing all with release builds of CMake with my global qbittorrent profile.

Reply:
> @glassez is this necessary?

I think no. In any case, it can easily be added temporarily if needed to test something.

Reply:
According to https://doc.qt.io/qt-5/qtglobal.html#qDebug
>This function does nothing if QT_NO_DEBUG_OUTPUT was defined during compilation.

Or @jagannatharjun you are actually using it?

Reply:
>you are actually using qDebug() for something?

yes, my personal builds include some extra feature I use (https://github.com/jagannatharjun/qBittorrent/tree/stream-2021)

also starting official qbittorrent v4.4.0 on top of master builds brings back a lot of deleted torrents, there are warnings in the console about QSQL missing 'download_path' field.

Reply:
> According to https://doc.qt.io/qt-5/qtglobal.html#qDebug
> 
> > This function does nothing if QT_NO_DEBUG_OUTPUT was defined during compilation.

Shouldn't QT_NO_DEBUG_OUTPUT defined in release mode?

Reply:
> Shouldn't QT_NO_DEBUG_OUTPUT defined in release mode?

QBittorrent has CMake code that manually adds QT_NO_DEBUG_OUTPUT, I had the code removed for testing purposes.

Reply:
> > Shouldn't QT_NO_DEBUG_OUTPUT defined in release mode?
> 
> QBittorrent has CMake code that manually adds QT_NO_DEBUG_OUTPUT, I had the code removed for testing purposes.

Well, then it's your choice.
Of course, I don't mind removing some annoying  and not so important debugging messages from the main codebase.